examples/eip712.rs

```diff
@@ -0,0 +1,125 @@
+use std::{convert::TryFrom, sync::Arc, time::Duration};
+
+use ethers::{
+    contract::EthAbiType,
+    prelude::*,
+    types::{transaction::eip712::Eip712, Address, I256, U256},
+    utils::{compile_and_launch_ganache, keccak256, Ganache, Solc},
+};
+use ethers_derive_eip712::*;
+
+#[derive(Debug, Clone, Eip712, EthAbiType)]
+#[eip712(
+    name = "Eip712Test",
+    version = "1",
+    chain_id = 1,
+    verifying_contract = "0x0000000000000000000000000000000000000001",
+    salt = "eip712-test-75F0CCte"
+)]
+struct FooBar {
+    foo: I256,
+    bar: U256,
+    fizz: Vec<u8>,
+    buzz: [u8; 32],
+    far: String,
+    out: Address,
+}
+
+abigen!(
+    DeriveEip712Test,
+    "./examples/derive_eip712_abi.json",
```

Comment:
This makes me realize we may want abigen to also work directly with solidity file so that it can generate both contract factories and just have easier UX

Reply:
I have a draft for that somewhere already that went stale a month ago or so lol, will revive that